groups-dev/createElements.js

```diff
@@ -0,0 +1,196 @@
+                <img class="card__count-icon" src="assets/person.svg" alt="Members" />
+            </span>
+        </div>
+    `;
+
+  if (group.isUpdating)
+    cardElement.querySelector('.card__btn').classList.add('button--blocked');
+  cardElement.querySelector('.card__title').textContent = group.title;
+  cardElement.querySelector('.card__description').textContent =
+    group.description;
+  cardElement.querySelector('.card__btn').textContent = group.isMember
+    ? 'Remove me'
+    : 'Add me';
+  if (group.isMember)
+    cardElement.querySelector('.card__btn').classList.add('button--secondary');
+  cardElement.querySelector('.card__count-text').textContent = group.count;
+  cardElement
+    .querySelector('.card__btn')
+    .addEventListener('click', () => group.isUpdating || onClick());
+
+  return cardElement;
+};
+
+const createLoadingCard = () => {
+  const cardElement = document.createElement('div');
+  cardElement.className = 'card card--loading';
+  cardElement.innerHTML = `
+        <div class="card__body">
+            <h5 class="card__title"></h5>
+            <p class="card__description"></p>
+        </div>
+        <div class="card__action">
+            <div class="card__btn"></div>
+        </div>
+    `;
+
+  return cardElement;
+};
+
+const createNavbarProfile = (profile) => {
+  const profileElement = document.createElement('div');
+  profileElement.className = 'profile';
+  profileElement.innerHTML = `
+        <h5 class="profile__name"></h5>
+        <img class="profile__avatar" />
+    `;
+
+  profileElement.querySelector(
+    '.profile__name',
+  ).textContent = `Hello, ${profile.name}`;
+  profileElement.querySelector(
+    '.profile__avatar',
+  ).alt = `Avatar of ${profile.name}`;
+  profileElement.querySelector('.profile__avatar').src = profile.avatar;
+
+  return profileElement;
+};
+
+const createNavbarProfileLoading = () => {
+  const profileElement = document.createElement('div');
+  profileElement.className = 'profile profile--loading';
+
+  return profileElement;
+};
+
+const createNavbarProfileSignin = () => {
+  const profileElement = document.createElement('a');
+  profileElement.className = 'signin';
+  profileElement.innerHTML = `
+      <span class="signin__text">Sign In With GitHub</span>
+      <img class="signin__img" src="assets/github.png" />
+    `;
+
+  profileElement.href = `https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97&}&state=${window.location.href}`;
+  return profileElement;
+};
+
+const createGroupCreationModal = (onClose = () => {}, onSubmit = () => {}) => {
+  const backdropElement = document.createElement('div');
+  const modalElement = document.createElement('div');
+  modalElement.className = 'group-creation-modal';
+  modalElement.innerHTML = `
+        <button type="button" id="close-button" class="button button--minimal">
+          <svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
```

Comment:
can we save them as svg and then use them here?

Reply:
Done